src/core/agent/services/signifyNotificationService.ts

```diff
@@ -151,14 +152,15 @@ class SignifyNotificationService extends AgentService {
   }
 
   async deleteNotificationRecordById(id: string): Promise<void> {
+    await this.markNotification(id);
     await this.notificationStorage.deleteById(id);
   }
 
   async processNotification(
     notif: Notification,
     callback: (event: KeriaNotification) => void
   ) {
-    if (notif.a.r === NotificationRoute.MultiSigRpy) {
+    if (notif.a.r === NotificationRoute.MultiSigRpy && !notif.r) {
```

Comment:
We should just have a simple
```
if (notif.r) {
  continue;
}
```
at the top. And then stop checking it everywhere else as we will miss somewhere.